src/iam/iam-base.ts

```diff
@@ -421,17 +404,19 @@ export class IAMBase {
   }
 
   private async setupNATS() {
-    if (this._natsServerUrl) {
+    const { messagingMethod, natsServerUrl } = this._messagingOptions || {};
+    if (natsServerUrl && messagingMethod === MessagingMethod.CacheServer) {
```

Comment:
If the conditional statement that is checking the MessagingMethod is in here, may we should change this method name to `setupMessaging`. What do you think?

Reply:
👍